### DISCUSSION TITLE
nested pipelines

### DISCUSSION THREAD

**Discussion by pcinereus** [ropensci/targets#1043] [General]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Is it possible to nest pipelines within pipelines.  For example, the section on Projects in the manual (https://books.ropensci.org/targets/projects.html) outlines how it is possible to configure and run multiple projects within the one root folder - however, is there a way to have an additional pipeline that runs these multiple (sub) pipelines?
Consider the following scenario:
Our team maintains a repo that contains many (>30) related quarto files (each with associated `knitr::read_chunk()` calls to populate chunks throughout the docs) .  Collectively, the series represents tutorials on a progression of statistical analyses ranging from simple regression up to complex spatio-temporal hierarchical models and multivariate analyses.
We would like to put a github actions push trigger in place to run any outdated Rscripts and compile any outdated upstream quarto documents and serve them as a site via github pages.  All the Rscripts and quarto documents reside in the one repo and we could declare each quarto (and associated R script(s)) as a "project", but then we would need to have some way to run each of these within the github actions workflow.  And if we then add another tutorial, there are a lot of moving parts to adjust.  Alternatively, we could have one overarching pipeline that includes all steps across all tutorials, however, this would be very large and unwieldy.
Ideally, it would be good to be able to construct nested pipelines defined across multiple files in order to manage the full dependency network.  Hope this makes sense....
Kind regards

Murray

**Comment by wlandau** [comment]
There are no plans to support nested pipelines, but you could group multiple reports into separate [Quarto projects](https://quarto.org/docs/projects/quarto-projects.html) and create a target for each project with [`tarchetypes::tar_quarto()`](https://docs.ropensci.org/tarchetypes/reference/tar_quarto.html). There are probably other ways to partition the appropriate amount of work into the appropriate number of targets.

A workflow like the one you describe can be a difficult fit because it is heavy on reports and scripts, whereas the `targets` package is focused on [function-oriented](https://books.ropensci.org/targets/functions.html) (rather than imperative/script-oriented) programming for data analysis. [Literate programming integration is supported](https://books.ropensci.org/targets/literate-programming.html), but it is usually used sparingly as a way to summarize a small number of results in a small number of short reports.

As I imply in https://books.ropensci.org/targets/targets.html, it can be worth pausing to reflect on the programming decisions for your specific situation: what a target should do, how much work it should do, what the return value should be, and what it is going to watch for changes so it can rerun at the appropriate time.

**Comment by pcinereus** [comment]
Thank you so much for you insights and I appreciate that my question was somewhat out of scope.  Your response has reassured me that my inability to track down a natural solution within `targets` alone is not the result of a lack of understanding or effort...
